packages/docusaurus/lib/commands/start.js

```diff
@@ -97,43 +98,40 @@ module.exports = async function start(siteDir, cliOptions = {}) {
     config = applyConfigureWebpack(configureWebpack, config, false);
   });
 
-  const compiler = webpack(config);
-
-  // Run webpack serve.
-  await serve(
-    {},
-    {
-      compiler,
-      open: true,
-      devMiddleware: {
-        logLevel: 'silent',
-      },
-      hotClient: {
-        port: hotPort,
-        logLevel: 'error',
-      },
-      logLevel: 'error',
-      port,
-      host,
-      add: app => {
-        // Serve static files.
-        const staticDir = path.resolve(siteDir, 'static');
-        if (fs.existsSync(staticDir)) {
-          app.use(mount(baseUrl, serveStatic(staticDir)));
-        }
-
-        // Enable HTTP range requests.
-        app.use(range);
-
-        // Rewrite request to `/` since dev is only a SPA.
-        app.use(
-          convert(
-            history({
-              rewrites: [{from: /\.html$/, to: '/'}],
-            }),
-          ),
-        );
-      },
+  // https://webpack.js.org/configuration/dev-server
+  const devServerConfig = {
+    compress: true,
+    clientLogLevel: 'error',
+    hot: true,
+    quiet: true,
+    headers: {
+      'access-control-allow-origin': '*',
+    },
+    publicPath: baseUrl,
+    watchOptions: {
+      ignored: /node_modules/,
     },
-  );
+    historyApiFallback: {
+      rewrites: [{from: /\.html$/, to: '/'}],
+    },
+    disableHostCheck: true,
+    overlay: false,
+    host,
+    contentBase: path.resolve(siteDir, 'static'),
```

Comment:
We could potentially add new hooks on this to the ability to execute Express.js middleware

https://webpack.js.org/configuration/dev-server/#devserverafter
https://webpack.js.org/configuration/dev-server/#devserverbefore

```js
before(app, server) {

},
after(app, server) {

}
```

Reply:
Sounds good! But let's add it when there's a request/we have a need. Gatsby's plugin lifecycles list is so long it's scary.